src/randoop/sequence/AbstractGenerator.java

```diff
@@ -155,34 +168,74 @@ public AbstractGenerator(List<Operation> operations, long timeMillis, int maxSeq
     }
     
     this.stopper = stopper;
-    
     this.listenerMgr = listenerManager;
-    
-    outputTestFilters = new LinkedList<ITestFilter>();
-    if (testfilters == null || testfilters.isEmpty()) {
-      outputTestFilters.add(new DefaultTestFilter());
-    } else {
-      outputTestFilters.addAll(testfilters);
+  }
+
+  /**
+   * Registers test predicate with the generator object for use while filtering
+   * generated tests.
+   * 
+   * @param outputTest  the predicate to be added to object
+   */
+  public void addTestPredicate(Predicate<ExecutableSequence> outputTest) {
+    if (outputTest == null) {
+      throw new IllegalArgumentException("outputTest must be non-null");
     }
+    this.outputTest = outputTest;
   }
 
+  /**
+   * Registers a visitor with this object for use while executing generated
+   * sequences.
+   * 
+   * @param executionVisitor  the visitor
+   */
+  public void addExecutionVisitor(ExecutionVisitor executionVisitor) {
+    if (executionVisitor == null) {
+      throw new IllegalArgumentException("executionVisitor must be non-null");
+    }
+    this.executionVisitor = executionVisitor;
+  }
+  
+  /**
+   * Registers a visitor with this object to generate checks following
+   * execution of generated tests.
+   * 
+   * @param checkGenerator  the check generating visitor 
+   */
+  public void addTestCheckGenerator(TestCheckGenerator checkGenerator) {
+    if (checkGenerator == null) {
+      throw new IllegalArgumentException("checkGenerator must be non-null");
+    }
+    this.checkGenerator = checkGenerator;
+  }
+  
   protected boolean stop() {
     return
     (listenerMgr != null && listenerMgr.stopGeneration())
     || (timer.getTimeElapsedMillis() >= maxTimeMillis)
-    || (numSequences() >= maxSequences)
+    || (numOutputSequences() >= maxOutputSequences)
+    || (numGeneratedSequences() >= maxGeneratedSequences)
     || (stopper != null && stopper.stop());
   }
 
   public abstract ExecutableSequence step();
 
-  public abstract int numSequences();
-
+  public int numOutputSequences() {
+    return outFailureSeqs.size() + outRegressionSeqs.size();
+  }
+  
+  public abstract int numGeneratedSequences();
+  
   /**
-   * Creates and executes new sequences in a loop.
+   * Creates and executes new sequences until stopping criteria is met.
+   * @see AbstractGenerator#stop()
+   * @see AbstractGenerator#step()
    */
   public void explore() {
-
+    if (checkGenerator == null) {
```

Comment:
To make sure you don't overlook any comments, review the comments at a URL like https://github.com/randoop/randoop/pull/73/files